_posts/2018-08-17-pxe-server-release.md

```diff
@@ -0,0 +1,153 @@
+---
+post_title: pxe-server release v1.0.0
+author: Piotr Król
+layout: post
+published: true
+post_date: 2018-08-17 16:00:00
+
+tags:
+	- pxe
+	- apu2
+	- coreboot
+categories:
+	- Firmware
+    - OS Dev
+---
+
+We mentioned many times that [pxe-server](https://github.com/3mdeb/pxe-server)
+helps us in embedded firmware development as well as in debugging various
+issues. `pxe-server` is part of our infrastructure and we keep maintaining it,
+since we rely on it during regression tests of PC Engines firmware.
+
+Recently I enter land of IOMMU and Xen, this lead me to need of Linux kernel as
+well as Xen kernel debugging and development. It happen that our `pxe-server`
+is not exactly prepared to help in seamless development, because of that we
+created `pxe-server-dev` which will serve as testing/development area before
+things can go to production `pxe-server`.
+
+Because of that I decided to publish instruction and all required scripts to
+provide more mature infrastructure for our regressions test and firmware
+development.
+
+# Proxmox
+
+Internally we have Proxmox setup on Dell PowerEdge R610, which we use for
+production `pxe-server` as well as `pxe-server-dev`. I started setup with
+`pxe-server-dev` since I need some place to work on streamlining of whole
+infrastructure, which were used in production.
+
+`pxe-server-dev`:
+* Memory: 2GiB
+* Processors: 2
+* Based on: template-debian-9.4.0-amd64-netinst (Linked clone)
+* HDD: 32GB
+* Networking: bridged
+
+`template-debian-9.4.0-amd64-netinst` is just plain netinstall without
+additional configuration, except:
+* SSH server which is required for ansible
+* `apt-transport-https` package
+* removing cdrom, which is initially required for installation iso
+
+Over time we should think about using `proxmox_kvm` to setup things. On the
+other hand I'm not huge fan of KVM and think Xen have much more sense, so in
+future we may plan to move away from Proxmox.
+
+# pxe-server release process
+
+So far there was no `pxe-server` release process we just pushed modifications
+to master and live with that. Whole `pxe-server` architecture is convoluted.
+First it is not just `pxe-server`, to be more precise it is TFTP and NFS
```

Comment:
it is HTTP and NFS now

Reply:
@miczyg1 yeah, completly forgot abut that.